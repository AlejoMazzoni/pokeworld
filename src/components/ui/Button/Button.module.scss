@import "/src/assets/css/media-querys.scss";

.buttonContainer {
  margin: 1rem;

  button {
    background-color: #51b8ec;
    font-weight: 600;
    padding: 0.8rem 2.7rem;
    border-radius: 1rem;
    border: none;
    cursor: pointer;

    @media (min-width: $tablet-s) {
      padding: 1.1rem 4rem;
      font-size: 1rem;
    }

    @media (min-width: $desktop-s) {
      padding: 1.1rem 4.5rem;
      font-size: 1.4rem;
    }
  }

  button:hover {
    transition-duration: 0.5s;
    background-color: rgba(125, 222, 235, 0.829);
    color: black;
  }
}
