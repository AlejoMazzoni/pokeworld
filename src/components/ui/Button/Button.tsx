import { ReactNode } from "react";
import styles from "./Button.module.scss";

interface IButton {
  children: ReactNode;
}

const Button = ({ children }: IButton) => {
  return (
    <div className={styles.buttonContainer}>
      <button>{children}</button>
    </div>
  );
};

export default Button;
