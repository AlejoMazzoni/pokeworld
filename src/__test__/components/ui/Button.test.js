import renderer from "react-test-renderer";
import Button from "../../../components/ui/Button";

describe("Button", () => {
  it("renders everything correctly", () => {
    const component = renderer.create(<Button>Click me</Button>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
