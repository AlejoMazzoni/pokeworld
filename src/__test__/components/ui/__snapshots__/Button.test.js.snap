// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders everything correctly 1`] = `
<div
  className="buttonContainer"
>
  <button>
    Click me
  </button>
</div>
`;
